app/electron/main.ts

```diff
@@ -421,7 +417,14 @@ function startElecron() {
 
     app.on('open-url', (event, url) => {
       mainWindow.focus();
-      dialog.showErrorBox('Welcome Back', `You arrived from: ${url}`);
+      const urlObj = new URL(url);
+      const authCode = urlObj.hash.split('&')[0];
```

Comment:
I think you should parse the URL here and check for the needed elements in a more robust way. (Unless the parse lib is less powerful than I thought and you tried it already)